src/hssm/param/params.py

```diff
@@ -0,0 +1,353 @@
+                param.is_parent = True
+                break
+
+        if self.parent == "":
+            for name, param in self.items():
+                if not param.is_fixed:
+                    self.parent = name
+                    param.is_parent = True
+                    break
+
+        if self.parent == "":
+            raise ValueError("No parent parameter found.")
+
+        self.parent_param = self[self.parent]
+
+    @classmethod
+    def from_user_specs(
+        cls,
+        model: HSSM,
+        include: list[dict[str, Any] | UserParam],
+        kwargs: dict[str, Any],
+        p_outlier: float | dict | bmb.Prior | None,
+    ) -> "Params":
+        """Create Params from user specifications.
+
+        Parameters
+        ----------
+        model
+            The HSSM model.
+        include
+            A list of dictionaries or UserParam objects specifying the parameters.
+        kwargs
+            Keyword arguments specifying the parameters.
+        p_outlier
+            The prior specification for the outlier probability.
+
+        Returns
+        -------
+        Params
+            The Params object with the specified parameters.
+        """
+        user_params = collect_user_params(model, include, kwargs, p_outlier)
+        params = make_params(model, user_params)
+        return cls(params)
+
+    def parse_bambi(
+        self,
+        model: HSSM,
+    ) -> tuple[bmb.Formula, dict | None, dict[str, str | bmb.Link] | str]:
+        """Retrieve three items that helps with bambi model building.
+
+        Returns
+        -------
+        tuple
+            1. A bmb.Formula object.
+            2. A dictionary of priors.
+            3. A dictionary of string or bmb.Link object representing the link
+            function.
+        """
+        # Handle the edge case where list_params is empty:
+        if not self.data:
+            return bmb.Formula(f"{model.response_c} ~ 1"), None, "identity"
+
+        parent_formula = None
+        other_formulas = []
+        priors: dict[str, Any] = {}
+        links: dict[str, str | bmb.Link] = {}
+
+        for param_name, param in self.items():
+            formula, prior, link = param.parse_bambi()
+
+            if param.is_parent:
+                # parent is not a regression
+                if formula is None:
+                    parent_formula = f"{model.response_c} ~ 1"
+                    if prior is not None:
+                        priors |= {param.name: {"Intercept": prior}}
+                    links[param_name] = "identity"
+                # parent is a regression
+                else:
+                    rhs = formula.split(" ~ ")[1]
+                    parent_formula = f"{model.response_c} ~ {rhs}"
+                    if prior is not None:
+                        priors[param_name] = prior
+                    if link is not None:
+                        links[param_name] = link
+            else:
+                # non-regression case
```

Comment:
maybe add explanation for why only non-regression case needs to be processed here.

Reply:
This is actually just a bad comment that wasn't deleted